src/opensignals/data/iexcloud.py

```diff
@@ -0,0 +1,102 @@
+import os
+
+import datetime as dt
+import time as _time
+from typing import Tuple
+from dateutil.relativedelta import relativedelta
+
+import numpy as np
+import pandas as pd
+from opensignals.data.provider import Provider
+from iexfinance import stocks
+from iexfinance.utils.exceptions import IEXQueryError
+
+
+class IEXCloud(Provider):
+    """Implementation of a stock dividend APY that uses IEXCloud"""
+
+    def __init__(self, iex_token: str = None, iex_api_version: str = 'iexcloud-sandbox', get_prices=True, get_dividends=True):
+        # iexcloud package expects IEX_TOKEN and IEX_API_VERSION environment variables
+        if "IEX_TOKEN" not in os.environ:
```

Comment:
From a quick look at the [iexfinance github readme](https://github.com/addisonlynch/iexfinance), it seems like you could pass the token to the library functions and also instance when instantiating the Stock object.